include/multipass/mount_handler.h

```diff
@@ -45,31 +43,61 @@ class MountHandler : private DisabledCopyMove
 
     virtual ~MountHandler() = default;
 
-    // Used to set up anything host side related
-    virtual void init_mount(VirtualMachine* vm, const std::string& target_path, const VMMount& vm_mount) = 0;
-    virtual void start_mount(VirtualMachine* vm, ServerVariant server, const std::string& target_path,
-                             const std::chrono::milliseconds& timeout = std::chrono::minutes(5)) = 0;
-    virtual void stop_mount(const std::string& instance, const std::string& path) = 0;
-    virtual void stop_all_mounts_for_instance(const std::string& instance) = 0;
-    virtual bool has_instance_already_mounted(const std::string& instance, const std::string& path) const = 0;
+    void start(ServerVariant server, std::chrono::milliseconds timeout = std::chrono::minutes(5))
+    {
+        if (active)
+            return;
+        start_impl(server, timeout);
+        active = true;
```

Comment:
Hmm, looking at the code, I don't think the `atomic_bool` is enough in here after all. I thought of `if (active.exchange(true))` instead, which would prevent concurrent starts from calling `start_impl` at the same time. However, without further care, that would also allow the state to be set to active when `start_impl` failed. And yet, it would not help us with mixed starts/stops, when executed concurrently. A lock is needed for that, AFAICS, at which point the atomic becomes redundant. 

As it is, the reader may get the impression that concurrency has been dealt with, upon seeing the atomic. So, if you don't feel like doing yet another significant iteration on the code to add a lock, I would ask for just a "TODO" comment making it clear that this still needs to be made thread safe.

Reply:
Ah, yes, I see what you mean. Would something like this solve it?
```cpp
void start(...) {
    std::lock_guard active_lock{active_mutex};
    if (active)
        return;
    start_impl(...);
    active = true;
}
...
private:
    bool active;
    std::mutex active_mutex;
```

Reply:
It would indeed, that would be perfect!

Reply:
Alright, done!